bin/civihr-install

```diff
@@ -0,0 +1,820 @@
+#!/bin/bash
+
+###############################################################################
+## Global variables
+
+## The location of the Project tree
+PRJDIR=
+
+## Location to store temp files, defaults (PRJDIR/tmp)
+TMPDIR=
+
+## Location that  points to sites that we build.
+BLDDIR=
+
+## Codename for the build instance.
+SITE_NAME=
+
+## Identifies site pointing ot default state.
+## (default: default)
+SITE_ID=default
+
+## Unique token for this site.
+## (default: random)
+SITE_TOKEN=
+
+## Root directory where the site's code will be installed
+## (default: BLDDIR/SITE_NAME)
+WEB_ROOT=
+
+## Root directory where we store cached copies of git repositories
+## (default: TMPDIR/git-cache)
+CACHE_DIR=
+
+## Time to wait before allowing updates to git/svn caches (seconds)
+CACHE_TTL=60
+
+## When updating a cache record, first attempt to lock it. Wait up to X seconds to acquire lock.
+CACHE_LOCK_WAIT=120
+
+## When checking out or updating git/svn, wait up to X seconds for process to complete
+SCM_TIMEOUT=3600
+
+## Whether the site has been previously installed
+IS_INSTALLED=
+
+## Default user accounts
+ADMIN_EMAIL="admin@example.com"
+ADMIN_PASS=
+ADMIN_USER="admin"
+
+## Printable name of the site (default: $SITE_NAME)
+CMS_TITLE=
+
+## The Drupal version
+CMS_VERSION=
+
+## The CiviCRM API (default: randomly generated)
+CIVI_SITE_KEY=
+
+## The CiviCRM branch/version
+CIVI_VERSION=4.7.9
+
+## Public URL of the site
+CMS_URL=
+
+## Path to the base of the CMS
+## (default: WEB_ROOT)
+CMS_ROOT=
+
+## DB credentials for CMS
+CMS_DB_DSN=
+CMS_DB_HOST=127.0.0.1
+CMS_DB_NAME=
+CMS_DB_PASS=
+CMS_DB_PORT=3306
+CMS_DB_USER=
+
+## Path to the civicrm-core tree
+CIVI_CORE=
+
+## DB credentials for Civi
+CIVI_DB_DSN=
+CIVI_DB_HOST=127.0.0.1
+CIVI_DB_NAME=
+CIVI_DB_PASS=
+CIVI_DB_PORT=3306
+CIVI_DB_USER=
+
+## Path to the civicrm.settings.php
+CIVI_SETTINGS=
+
+## Path to the civicrm files directory
+CIVI_FILES=
+
+## Path to the civicrm templates_c cache directory
+CIVI_TEMPLATEC=
+
+## Name of the CiviCRM UF (Drupal)
+CIVI_UF=
+
+## The name of the organization (for inclusion in footers, etc)
+CIVI_DOMAIN_NAME=
+
+## The default from email address
+CIVI_DOMAIN_EMAIL=
+
+## Path to the web-managed extension folder (optional)
+CIVI_EXT_DIR=
+
+## URL of the web-managed extension folder (required if CIVI_EXT_DIR is set)
+CIVI_EXT_URL=
+
+
+
+###############################################################################
+## Display usage message
+function civihr_app_usage() {
+  APP=$(basename "$0")
+
+  php -r 'echo file_get_contents("php://stdin");' <<EOT
+
+Common Options :
+  <site-name>        The name of the site-directory to build
+
+Syntax: $APP <site-name> [required-options]
+Description: Downloads and/or installs the application.
+
+Required options:
+  --dbuser <mysql-username>     Username for mysql
+  --dbpass <mysql-password>     Password for mysql
```

Comment:
Why is the password required? What if I want to use an user that doesn't have a password?